Added error messages for RH_RF95 init

scottbarnesg opened the pull request:
Fixes Issue #31: Insufficient error messages on RH_RF95 init.

Adds error messages for each failure (`return false`) condition of the RH_RF95 init function. This will make debugging initialization issues significantly easier for users.

ladyada commented:
Serial.println()'ing from a library is considered bad taste - you may want to, instead, have an #ifdef that people can turn on for more debugging - that's a much better ideas! also please use F("string") so AVRs dont use RAM for strings. thanks!

scottbarnesg commented:
Updated per @ladyada's feedback:
- added `DEBUG_SERIAL` definition (commented out)
- added `ifdef` around all Serial.print statements
- added `F("string")` for eligible Serial.print statements (exception: line 74 not defined until run time).
- added print statement in rf95 failure condition for feather rf95 examples that use RH_RF95 class  that describes how to enable debug mode.

Another option would be to update the RH_RF95 class with optional debug parameter that defaults to false, but I wasn't sure if this would cause compilation issues with environments that don't support `Serial`. Are there any target environments that don't support `Serial`? If not, I think this would be the better option.

ladyada commented:
nice thank you :)
